### ISSUE TITLE
bad interaction with rust-toolchain.toml

### ISSUE THREAD

**Issue by pyaillet** [esp-rs/xtensa-toolchain#2]
Hi !

Thank you for this gh action, I was searching exactly for this :)

However I faced a problem when trying to use it on one of my project. Not sure if this is an issue but I thought it was worth reporting here.

I created a rust-toolchain.toml file with the following content to build my project locally without messing with the default toolchain of my system.
```toml
[toolchain]

channel = "esp"
```

However, when I used the gh action with it I got the following error:

```
Run esp-rs/xtensa-toolchain@v1
Run actions-rs/toolchain@v1
/home/runner/.cargo/bin/rustup show
Default host: x86_64-unknown-linux-gnu
rustup home:  /home/runner/.rustup

(error: invalid toolchain name: 'esp')
/home/runner/.cargo/bin/rustup -V
rustup 1.24.3 (ce5817a94 2021-05-31)
info: This is the version for the rustup toolchain manager, not the rustc compiler.
Installed rustup 1.24.3 support profiles
/home/runner/.cargo/bin/rustup set profile minimal
info: profile set to 'minimal'
/home/runner/.cargo/bin/rustup toolchain install stable
info: syncing channel updates for 'stable-x86_64-unknown-linux-gnu'

  stable-x86_64-unknown-linux-gnu unchanged - rustc 1.59.0 (9d1b2106e 2022-02-23)

info: checking for self-updates
warning: tool `rustfmt` is already installed, remove it from `/home/runner/.cargo/bin`, then run `rustup update` to have rustup manage this tool.
warning: tool `cargo-fmt` is already installed, remove it from `/home/runner/.cargo/bin`, then run `rustup update` to have rustup manage this tool.
/home/runner/.cargo/bin/rustup default stable
info: using existing install for 'stable-x86_64-unknown-linux-gnu'
info: default toolchain set to 'stable-x86_64-unknown-linux-gnu'

  stable-x86_64-unknown-linux-gnu unchanged - rustc 1.59.0 (9d1b2106e 2022-02-23)

error: invalid toolchain name: 'esp'
```

If I delete the `rust-toolchain.toml` on the CI before launching the action it [fixes the problem](https://github.com/pyaillet/twatch-idf-rs/runs/5431824647?check_suite_focus=true) but seems really "hackish".

Here is my workflow:
```yaml
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: Remove rust-toolchain.toml
        run: rm rust-toolchain.toml
      - name: Install Rust for Xtensa
        uses: esp-rs/xtensa-toolchain@v1
        with:
          default: true
          ldproxy: true
      - name: Release build std
        uses: actions-rs/cargo@v1
        with:
          command: build
```

**Comment by jessebraham** [comment]
Thanks for opening this issue, I actually just ran into this myself recently. I will try to get this fixed ASAP.

**Comment by jessebraham** [comment]
I do not plan on creating a new release today, but for the time being you should be able to use the commit hash below which seems to have resolved the problem.

```text
uses: esp-rs/xtensa-toolchain@9de5c646b53804bb3d042eeae6cbd9a4fd433dd9
```

I will leave this issue open until a new version is released.

**Comment by pyaillet** [comment]
The new version [worked for me](https://github.com/pyaillet/twatch-idf-rs/actions/runs/1946908406).

Thanks !

**Comment by jessebraham** [comment]
Sort of forgot about this (oops 😁 ) but I have now released `v1.1` which includes the fix for this issue.

**Comment by pyaillet** [comment]
No problem :)

Thank you very much for the release !
